File: benchmark/corpus/01_file_write/b_two_levels.js
// The write sits two calls down: main -> summarise -> persist.
function persist(path, text) {
  Deno.writeTextFileSync(path, text); // DANGER
}

function summarise(path, amounts) {
  const total = amounts.reduce((a, b) => a + b, 0);
  persist(path, "total " + total);
  return total;
}

const path = (await new Response(Deno.stdin.readable).text()).trim();
console.log(summarise(path, [2500, 45000, 12000]));

File: benchmark/corpus/01_file_write/b_two_levels.py
# The write sits two calls down: main -> summarise -> persist.
import sys


def persist(path, text):
    open(path, "w").write(text)  # DANGER


def summarise(path, amounts):
    total = sum(amounts)
    persist(path, "total %d" % total)
    return total


path = sys.stdin.readline().strip()
print(summarise(path, [2500, 45000, 12000]))
